### ISSUE TITLE
Adjust Mismatches when using JASPAR motifs

### ISSUE THREAD

**Issue by JavBurgoa** [loosolab/TOBIAS#17]
Hi!, I have an issue when running BINDetect with my own motif JASPAR file. Each motif within the JASPAR file looks like this:

```
>GGGTCAAGTTCA  DR0-10
A [ 0 0 0 0 0 100 100 0 0 0 0 100]
C [ 0 0 0 0 100 0 0 0 0 0 100 0]
G [ 100 100 100 0 0 0 0 100 0 0 0 0]
T [ 0 0 0 100 0 0 0 0 100 100 0 0]
```
Where each number is suposed to represent the proportion of a nucleotide happening in each position of the motif. 
Theoretically, with this disposition BINDetect should allow 0 mismatches when looking for occurrences of this motif in the genome. However, mismatches happen. The way I found out that there were actual mismatches was using the _bound.bed bedfile for this motif, getting various randomly selected lines from it and finding out which sequence they actually correspond to by means of bedtools' tool getfasta and the original genome I used for the whole analysis:

```
bedtools getfasta -bed GGGTCAAGTTCA_sample_bound.bed -fasta Genome.fasta
```
The sequences I get do not match exactly my original motif:

Original Motif | GGGTCAAGTTCA
------------------------------------
Getfasta result 1 | GGG**A**CAAG**C**TCA
Getfasta result 2 | GGGTCAAGTT**AT**

Am I looking in the wrong file (_bound.bed file is not what I think it is), or am I doing something wrong when formatting my JASPAR file? Thanks a lot!

**Comment by msbentsen** [comment]
Hi! This is a very valid question, and I will try to explain the best I can below. What happens before the motif scanning is that the PFM is converted into a PSSM (position-specific scoring matrix). This takes the background GC-content into account, and also adds a pseudocount for the 0-counts. So for a peak set with GC content 47%, the PSSM for your motif is calculated as:
```
A -4.62 -4.62 -4.62 -4.62 -4.62  1.33  1.33 -4.61 -4.61 -4.61 -4.61  1.33 
C -4.62 -4.62 -4.62 -4.62  1.43 -4.62 -4.62 -4.62 -4.62 -4.62  1.43 -4.62
G  1.43  1.43  1.43 -4.62 -4.62 -4.62 -4.62  1.43 -4.62 -4.62 -4.62 -4.62 
T -4.62 -4.62 -4.62  1.33 -4.62 -4.62 -4.62 -4.62  1.33  1.33 -4.62 -4.62
```
So a '0' in the motif PFM does not mean exclusion of this nucleotide from the motif, but rather that it is not very likely to occur (very low scores in the scoring matrix). The threshold for whether a sequence is a hit towards the motif is determined by the `--motif-pvalue` option in `BINDetect`, and you can change this to make the motifs more/less stringent with regards to mismatches.

**So tl;dr:** If you set the motif p-value for `BINDetect` very low (like `--motif-pvalue 0.00000001`), you can control that you only get exact matches. Hope this answered your question!

**Comment by JavBurgoa** [comment]
Hi, yes, that completely solves my problem. Very quick, good explanation. Thanks!
